### ISSUE TITLE
Generate report now

### ISSUE THREAD

**Issue by toddsby** [martomi/chiadog#106]
I've successfully installed and started chiadog but it appears I have to wait 24 hours to get a report? My full-node "farmer" has been running for days and the debug logs should be full. Why is there no way to generate a report immediately if I already have full logs?? or am I missing something. @martomi

**Comment by pieterhelsen** [comment]
Thanks for your feedback.
Chiadog starts to collect data from the time you start the program. It does not go back in time to analyze past logs. 
As such, theoretically, if you would create a report immediately after startup, all values would be zero or empty either way.

**Comment by toddsby** [comment]
> Chiadog starts to collect data from the time you start the program. It does not go back in time to analyze past logs.
Why not? Why do I have to wait 24 hours for it provide value?

**Comment by pieterhelsen** [comment]
Well, primarily because it has not been implemented that way. The rationale behind it only Martin will be able to give definitively, but here's my take on it:

- Chiadog is a watchdog; watchdogs generally monitor current events and do not concern themselves with past events. Daily stats, as such is even expanding on the scope of a 'pure' watchdog. 
- Assumedly, most people will not have their log levels set to INFO before they install Chiadog, as such, a similar functionality would be of little use to a large percentage of the user base
- Processing 20+ MBs of data is processing-intensive and is something that should be done intentionally, not as a part of the startup process of Chiadog
- File rotation complicates this process further
- You're asking for previous 24h analysis, the next guy will ask for previous 2 days, previous 10 hours, previous week... 

The bottom line... I would expect this feature is best suited for a separate project (maybe name it `Chialog`)

**Comment by martomi** [comment]
What Pieter said 😃 I see a value and place for such a tool but this project focuses only on real-time monitoring.
